dogen/tools.py

```diff
@@ -1,26 +1,93 @@
+import hashlib
 import logging
 import os
 import requests
-import subprocess
-import urllib
 
 from dogen.errors import DogenError
 
 
-logger = logging.getLogger("dogen")
+SUPPORTED_HASH_ALGORITHMS = ['sha256', 'sha1', 'md5']
+logger = logging.getLogger('dogen')
 
 
-def fetch_artifact(url, directory=os.getcwd(), ssl_verify=True):
-    destination = os.path.join(directory, os.path.basename(url))
-    logger.debug("Fetching '%s' as %s" % (url, destination))
+class Artifact(object):
+    """A class representing artifact """
+    ssl_verify = True
+    check_integrity = True
+    target_dir = ""
```

Comment:
This target directory (set somewhere else) is hard to track. When we look at the object alone we cannot see it set anywhere, we need to look for setting it to the class somewhere else. Whats the benefit of such code?

For me it should be part of initialization of the object.